ui/src/widgets/ui-markdown/UIMarkdown.vue

```diff
@@ -1,51 +1,126 @@
-<template>
-    <vue-markdown class="nrdb-ui-widget nrdb-ui-markdown" :source="props.content" />
-</template>
-
 <script>
-import VueMarkdown from 'vue-markdown-render'
-import { mapState } from 'vuex'
+import DOMPurify from 'dompurify'
+import hljs from 'highlight.js'
+import 'highlight.js/styles/default.css'
+import { marked } from 'marked'
+import { markedHighlight } from 'marked-highlight'
+
+import { h } from 'vue'
+import { mapGetters, mapState } from 'vuex' // eslint-disable-line import/order
 
 import { useDataTracker } from '../data-tracker.js'
 
+marked.use(markedHighlight({
+    langPrefix: 'hljs language-',
+    highlight (code, lang) {
+        const language = hljs.getLanguage(lang) ? lang : 'plaintext'
+        return hljs.highlight(code, { language }).value
+    }
+}))
+
 export default {
     name: 'DBUIMarkdown',
-    components: {
-        VueMarkdown
-    },
     inject: ['$socket'],
     props: {
         id: { type: String, required: true },
         props: { type: Object, default: () => ({}) }
     },
+    setup (props) {
+        useDataTracker(props.id)
```

Comment:
why the need for a nested component here? Can't all of these values/functions live at the `DBUIMarkdown` component level?

Reply:
it was nested similar to the template so that dynamic bindings would work. 

Or am I misunderstanding your question?

Reply:
Have tried to do an `npm install` and `npm run build` and getting:

```
 ERROR  Failed to compile with 1 error                                                                                                                                                                                                          11:45:14 AM

[eslint] 
/Users/joepavitt/Documents/flowforge/development/nodes/flowforge-nr-dashboard/ui/src/widgets/ui-markdown/UIMarkdown.vue
  2:23  error  Unable to resolve path to module 'dompurify'         import/no-unresolved
  2:23  error  "dompurify" is not found                             n/no-missing-import
  5:24  error  Unable to resolve path to module 'marked'            import/no-unresolved
  5:24  error  "marked" is not found                                n/no-missing-import
  6:33  error  Unable to resolve path to module 'marked-highlight'  import/no-unresolved
  6:33  error  "marked-highlight" is not found                      n/no-missing-import
```

which is odd, because they are in the `devDependencies`

Reply:
deleting `/node_modules` and re-doing `npm install` unblocked it.